### ISSUE TITLE
problem markers auto-created by tasks extension are not auto-removed after problems being fixed

### ISSUE THREAD

**Issue by elaihau** [theia-ide/theia#5352]
### Description

This is a follow-up issue for pull request https://github.com/theia-ide/theia/pull/5024.

With the patch of https://github.com/theia-ide/theia/pull/5024, users are able to define problem matchers and problem patterns to parse the task output, and have problem markers automatically created in the problems view.

However, those auto-created problem markers won't be auto-removed when problems are fixed, unless users run the same task one more time.

### Reproduction Steps

Run a task with a problem matcher defined in `tasks.json`.
Take `npm:lint` for example,

- you can have the following in the tasks.json to create problem markers:
```json
        {
            "type": "npm",
            "script": "lint",
            "problemMatcher": {
                "owner": "eslint",
                "source": "eslint",
                "applyTo": "alldocuments",
                "fileLocation": ["absolute"],
                "pattern": [{
                    "regexp": "^([^\\s].*)$",
                    "kind": "file",
                    "file": 1
                },
                {
                    "regexp": "^\\s+(\\d+):(\\d+)\\s+(error|warning|info)\\s+(.+?)(?:\\s\\s+(.*))?$",
                    "line": 1,
                    "character": 2,
                    "severity": 3,
                    "message": 4,
                    "code": 5,
                    "loop": true
                }]
            }
        }
```
- Run npm:lint as a task in Theia. If there are any lint errors, they should be displayed in the problems view as error markers. 
- Go to the line that has an error, and fix that error
- Check the problem view.

What happens: Problems view has the same markers for eslint errors.
What should happen: The marker for the error that was just fixed should not be in the problems view. 

**OS and Theia version:**

Ubuntu 16.04. Example browser that runs Theia 0.7.0 with the patch of https://github.com/theia-ide/theia/pull/5024.

**Comment by marcdumais-work** [comment]
Is this the behavior in VS Code? I thought that problem markers created by a task were only removed when the task is run again (and then they are potentially re-created by the latest task run, if they are still applicable).

**Comment by elaihau** [comment]
> Is this the behavior in VS Code? I thought that problem markers created by a task were only removed when the task is run again (and then they are potentially re-created by the latest task run, if they are still applicable).

@marcdumais-work 

I would say the answer is "depends".

This is how vscode behaves regarding the "eslint problems" found by the problem matcher "$eslint-stylish". As you can see, I can double click the markers in the problems view to open the problematic line in the editor. And the problem marker gets removed after the editor is closed even if the problem is not fixed.

![Peek 2020-02-04 21-53](https://user-images.githubusercontent.com/37082801/73806945-e624e480-4798-11ea-9a2c-c281fcd4a654.gif)


Let's see the GIF below for another example.
Same npm task, same problem matcher, but this time the problem matcher is associated with different owner and source (i.e., "eslint999").

After the problem markers are created, vscode is not able to remove them. As you can see, even if I opened and closed the editor, the problem markers stayed in the problems view.

One thing I forgot to show in the GIF was, even if I opened the editor, manually fixed the eslint warining in the editor, the problem markers stayed.

![Peek 2020-02-04 21-58](https://user-images.githubusercontent.com/37082801/73807160-80852800-4799-11ea-83ab-d1cf71fa3f37.gif)


So the conclusion from running the tests above is, theoretically the behavior is NOT in vscode. However in some scenarios, vscode and its extensions give users "illusions" that the problem markers created by the problem matchers are taken care of, after the problems are fixed.

**Comment by marcdumais-work** [comment]
Thanks @elaihau 

Looking into tangentially related VS Code issues (see below), I think that having task-generated problems automatically cleared from the IDE, after being fixed, can only practically happen for special cases (e.g. task that runs a watcher, that will react to code changes, recompile the file and send a fresh set of diagnostics to the IDE). 

TL;DR I do not see a generic way for a task to clear the problem markers it generated, without re-running the same task. 

https://github.com/microsoft/vscode/issues/47386
https://github.com/microsoft/vscode/issues/50448
https://github.com/microsoft/vscode/issues/58088
https://github.com/microsoft/vscode/issues/59363
